### PR TITLE
Update installation section for new release

### PR DESCRIPTION
**Pull request by nd-02110114**
## Description

This PR is a part of  #2343 

I update installation section in README.md and docs.
And, I updated Dockerfile and pushed the new image to dockerhub.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/__init__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 """
 
 # If you push the tag, please remove `.dev`
-__version__ = '2.4.0'
+__version__ = '2.4.1.dev'
```
Comment (nd-02110114) [inline deepchem/__init__.py]: We have already pushed the 2.4.0 tag, so I set the dev version of the next release.

File: deepchem/__init__.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 """
 
 # If you push the tag, please remove `.dev`
-__version__ = '2.4.0'
+__version__ = '2.5.0.dev'
```
Comment (nd-02110114) [inline deepchem/__init__.py:6]: We have already pushed the 2.4.0 tag, so I set the dev version of the next release.

**Comment by nd-02110114** [comment]
And then, I published the documentation for 2.4.0 version.
https://deepchem.readthedocs.io/en/2.4.0/
https://readthedocs.org/projects/deepchem/versions/

I also tried publishing the 2.3.0 version for backward compatibility.
However, the docs build failed and I gave up...

**Review by rbharath — APPROVED** [review]
Thank you for the update! Going to go ahead and merge this in :)

**Pull request merged**